ConsumerApi.Tests.Integration/StepDefinitions/IdentitiesApiStepDefinitions.cs

```diff
@@ -0,0 +1,96 @@
+using Backbone.ConsumerApi.Tests.Integration.Extensions;
+using Backbone.ConsumerApi.Tests.Integration.Models;
+using Backbone.Crypto;
+using Backbone.Crypto.Implementations;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Backbone.ConsumerApi.Tests.Integration.StepDefinitions;
+
+[Binding]
+[Scope(Feature = "POST Identity")]
+internal class IdentitiesApiStepDefinitions : BaseStepDefinitions
+{
+    private readonly ChallengesApi _challengeApi;
+    private readonly IdentitiesApi _identitiesApi;
+    private HttpResponse<CreateIdentityResponse>? _identityResponse;
+    private HttpResponse<Challenge>? _challengeResponse;
+
+    public IdentitiesApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, IdentitiesApi identitiesApi, ChallengesApi challengeApi) : base(httpConfiguration)
+    {
+        _identitiesApi = identitiesApi;
+        _challengeApi = challengeApi;
+    }
+
+    [Given(@"a Challenge c")]
+    public async Task GivenAChallengeC()
+    {
+        _challengeResponse = await _challengeApi.CreateChallenge(_requestConfiguration);
+        _challengeResponse!.IsSuccessStatusCode.Should().BeTrue();
+    }
+
+    [When(@"a POST request is sent to the /Identities endpoint")]
+    public async Task WhenAPOSTRequestIsSentToTheIdentitiesEndpoint()
+    {
+        var serializedChallenge = JsonConvert.SerializeObject(new
+        {
+            _challengeResponse!.Content.Result!.Id,
+            _challengeResponse!.Content.Result!.ExpiresAt,
+            Type = "Identity"
+        });
+
+        var signatureHelper = SignatureHelper.CreateEd25519WithRawKeyFormat();
+        var keyPair = signatureHelper.CreateKeyPair();
+        var signature = signatureHelper.CreateSignature(ConvertibleString.FromUtf8(serializedChallenge), keyPair.PrivateKey);
+
+        dynamic publicKey = new
+        {
+            pub = keyPair.PublicKey.Base64Representation,
+            alg = 3
+        };
+
+        dynamic signedChallenge = new
+        {
+            sig = signature.BytesRepresentation,
+            alg = 2
+        };
+
+        var createIdentityRequest = new CreateIdentityRequest()
+        {
+            ClientId = "test",
+            ClientSecret = "test",
+            DevicePassword = "test",
+            IdentityPublicKey = (ConvertibleString.FromUtf8(JsonConvert.SerializeObject(publicKey)) as ConvertibleString)!.Base64Representation,
+            IdentityVersion = 1,
+            SignedChallenge = new CreateIdentityRequestSignedChallenge()
+            {
+                Challenge = serializedChallenge,
+                Signature = (ConvertibleString.FromUtf8(JsonConvert.SerializeObject(signedChallenge)) as ConvertibleString)!.Base64Representation
+            }
+        };
+
+        var requestConfiguration = _requestConfiguration.Clone();
+        requestConfiguration.ContentType = "application/json";
+        requestConfiguration.SetContent(createIdentityRequest);
```

Comment:
Can you create an overload of the `CreateIdentity` method in the `IdentitiesApi` without the RequestConfiguration parameter that does all of this + the creation of the challenge? Because the goal of your task was to make it possible to create identities in other tests. While the goal is theoretically achieved, what do you imagine the tests will look like? Are you going to copy&paste all this code over and over again?

Reply:
You could simply use the exact same steps in the tests, thought that was exactly the point.

Reply:
Ahm... You're right, I'm stupid, let's forget about this comment. :D